src/SmartComponents/BaselinesPage/EditBaseline/EditBaseline.js

```diff
@@ -236,30 +284,50 @@ export class EditBaseline extends Component {
                 rowData = this.renderRowData(facts[i]);
                 rows.push(rowData);
             }
-        } else {
-            rows =
-                <td colSpan='3'>
-                    <EmptyStateDisplay
-                        title={ 'No facts' }
-                        text={ [ 'No facts or categories have been added to this baseline yet.' ] }
-                        button={ <AddFactButton /> }
-                    />
-                </td>;
         }
 
         return rows;
     }
 
+    renderEmptyState() {
```

Comment:
Have you thought of extracting this into a separate component?
It looks very similar to what is on https://github.com/RedHatInsights/drift-frontend/pull/341/files#diff-ff14a549333021d3958a55730a76b8dbR107 and maybe could be used in both places.

Reply:
Yeah, I meant creating another component to handle which EmptyStateDisplay to show. But looking at it closer it seems to just differ in the props it needs.

In compliance we try to avoid having functions in components that return components and prefer to extract them into a separate (functional) component. For readability and since we have seen components built that way act unexpected.